osidb/models/flaw/flaw.py

```diff
@@ -1063,9 +1079,9 @@ def _create_new_flaw():
                 return
 
             # we're handling an existing OSIDB-authored flaw -- update
-            if force_update or any(
-                getattr(old_flaw, field) != getattr(self, field)
-                for field in SYNC_REQUIRED_FIELDS
+            if force_update or (
+                diff is not None
+                and any(field in diff.keys() for field in SYNC_REQUIRED_FIELDS)
```

Comment:
What is the exact reason for using the new `diff` here but still using the `old_flaw` instance on the previous lines?